### PR TITLE
Created power_of_n.py

### PR DESCRIPTION
**Pull request by HimanshuGupta-p1**
Created a program that calculates x raised to the power of n, where x can be a float number or integer and n can be a positive or negative integer.

Example 1:

Input: x = 2.00000, n = 10
Output: 1024.00000
Example 2:

Input: x = 2.10000, n = 3
Output: 9.26100
Example 3:

Input: x = 2.00000, n = -2
Output: 0.25000
Explanation: 2^-2 = 1/(2^2) = 1/4 = 0.25

### CODE REVIEW

File: power_of_n.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+# Input: x = 2.10000, n = 3
+# Output: 9.26100
+# Example 3:
+
+# Input: x = 2.00000, n = -2
+# Output: 0.25000
+# Explanation: 2^-2 = 1/(2^2) = 1/4 = 0.25
+
+#Class
```
Comment (NitkarshChourasia) [inline power_of_n.py:24]: Leave a space after '#'.

File: power_of_n.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
+#Python program to calculate x raised to the power n (i.e., x^n)
+
+# Script Name		: power_of_n.py
```
Comment (NitkarshChourasia) [inline power_of_n.py:3]: Use """ docstring """ instead of comments.

File: power_of_n.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+# Author		    : Himanshu Gupta
```
Comment (NitkarshChourasia) [inline power_of_n.py:4]:
```python
# use 
__author__ == "Himanshu Gupta"

# try printing it.

print(__author__)
```

File: power_of_n.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+# Output: 9.26100
+# Example 3:
+
+# Input: x = 2.00000, n = -2
+# Output: 0.25000
+# Explanation: 2^-2 = 1/(2^2) = 1/4 = 0.25
+
+#Class 
+class Solution:
+
```
Comment (NitkarshChourasia) [inline power_of_n.py:26]: Also using docstring in class like this: to explain the brief task the program does, would be helpful.

```python
""" To output a given number to the power of n """

# if this is used for printing, see what happens.
# This is inbuilt dunder method.


print(function_name.__docs__)
```

File: power_of_n.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        return result
+
+
+obj = Solution() #Creating object of the class Solution
+
+#Taking inouts from the user
+x = float(input("Enter the base number: "))
+n = int(input("Enter the power number: "))
+
+#calling the function using object obj to calculate the power
```
Comment (NitkarshChourasia) [inline power_of_n.py:55]:
```python
# use this for executing the program when ran directly.

if __name__ == "__main__":
        print(function_name(param1, param2))
        
```

Comment (NitkarshChourasia) [reply power_of_n.py:55]: The equivalent is javascript would be...

```javascript

if (require.main == module) {
    console.log(function_name(param1))
    
```

File: power_of_n.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+answer = obj.binaryExponentiation(x, n)
+print(answer) #answer
```
Comment (NitkarshChourasia) [inline power_of_n.py:57]: # PLEASE DO ALL THIS CHANGES BEFORE MERGING, RESPECTFULLY.

**Review by NitkarshChourasia — CHANGES_REQUESTED** [review]

File: power_of_n.py (hunk 6 above)
Comment (HimanshuGupta-p1) [reply power_of_n.py]: Okay !.

File: power_of_n.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    Example 1:
+
+    Input: x = 2.00000, n = 10
+    Output: 1024.0
+    Example 2:
+
+    Input: x = 2.10000, n = 3
+    Output: 9.261000000000001
+
+    Example 3:
```
Comment (NitkarshChourasia) [inline power_of_n.py:19]: All perfect!

**Review by NitkarshChourasia — COMMENTED** [review]
All perfectly done.

**Pull request closed without merging**